DNN Platform/Library/Entities/Users/UserEventHandlers.cs

```diff
@@ -51,6 +51,7 @@ public void UserDeleted(object sender, UserEventArgs args)
 
         public void UserApproved(object sender, UserEventArgs args)
         {
+            if (!args.SendNotification) return;
             Mail.SendMail(args.User, MessageType.UserRegistrationPublic, PortalSettings.Current);
             DeleteAllNotifications(args.User.UserID);
```

Comment:
There is a `return` statement, so `DeleteAllNotifications` won't get called if `args.SendNotification` is `false`.